tasks/main.yml

```diff
@@ -198,41 +224,373 @@
+            dest: "{{ __hpc_hpcx_rebuild_path }}"
+            mode: "0755"
+            owner: root
+            group: root
+
+        - name: Remove extracted tarball
+          file:
+            path: "{{ __hpc_pkg_extracted.path }}"
+            state: absent
+          changed_when: false
+
+    - name: Ensure that /etc/lmod directory exists
+      file:
+        path: /etc/lmod
+        state: directory
+        owner: root
+        group: root
+        mode: '0755'
+
+    - name: Ensure that /etc/lmod/.modulespath file exists
+      file:
+        path: /etc/lmod/.modulespath
+        state: touch
+        owner: root
+        group: root
+        mode: '0755'
+
+    # This requires a new login shell to become visible.
+    - name: Set up hpcx module files for lmod
+      lineinfile:
+        path: /etc/lmod/.modulespath
+        line: "{{ __hpc_hpcx_rebuild_path }}/modulefiles"
+        state: present
+        owner: root
+        group: root
+        mode: '0755'
+
+    - name: Get stat of openmpi path
+      stat:
+        path: "{{ __hpc_openmpi_path }}"
+      register: __hpc_openmpi_path_stat
+
+    - name: Download and build OpenMPI
+      when: not __hpc_openmpi_path_stat.stat.exists
+      block:
+        - name: Download {{ __hpc_openmpi_info.name }}
+          include_tasks: tasks/download_extract_package.yml
+          vars:
+            __hpc_pkg_info: "{{ __hpc_openmpi_info }}"
+
+        - name: Build {{ __hpc_openmpi_info.name }}
+          command:
+            cmd: "{{ item }}"
+            chdir: "{{ __hpc_pkg_extracted.path }}"
+          changed_when: true
+          loop:
+            - >-
+              ./configure --prefix={{ __hpc_openmpi_path }}
+              --with-ucx={{ __hpc_ucx_path }}
+              --with-ucc={{ __hpc_ucc_path }}
+              --with-hcoll={{ __hpc_hcoll_path }}
+              --with-pmix={{ __hpc_pmix_path }}
+              --enable-prte-prefix-by-default
+              --with-platform=contrib/platform/mellanox/optimized
+              --with-cuda={{ __hpc_cuda_path }}
+            - make -j {{ ansible_processor_nproc }}
+            - make install
+
+        - name: Remove extracted tarball
+          file:
+            path: "{{ __hpc_pkg_extracted.path }}"
+            state: absent
+          changed_when: false
+
+    - name: Ensure MPI modulefile directory exists
+      file:
+        path: "{{ __hpc_module_dir }}/mpi"
+        state: directory
+        owner: root
+        group: root
+        mode: '0755'
+
+    - name: Install OpenMPI modulefile
+      template:
+        src: openmpi-ver.lua.j2
+        dest: "{{ __hpc_module_dir }}/mpi/openmpi-{{ __hpc_openmpi_info.version }}.lua"
+        owner: root
+        group: root
+        mode: '0644'
+
+- name: Tune system for HPC
+  when: hpc_tuning
+  block:
+    - name: Remove user memory limits to ensure applications aren't restricted
+      lineinfile:
+        path: /etc/security/limits.conf
```

Comment:
There is a /etc/security/limits.d/ directory for drop in files, so consider putting these in a file (or template if you want to include the ansible managed header and the system_role:hpc fingerprint).

Reply:
Done